### ISSUE TITLE
User Authentication Does Not Appear To Be Functional in 2.10.0

### ISSUE THREAD

**Issue by cashank** [zaproxy/zaproxy#6480]
**Describe the bug**
We've recently updated to 2.10.0 of ZAP; from 2.9.0, and have run into an issue with none of our authenticated users actually authenticating during active or passive spidering attempts.

Further investigation reveals that the default values (eg: email: foo-bar@example.com) are being put into the requests, despite those default values being deleted from the session and new ones are being explicitly called.

Possibly related: UI behaves differently at this step too, maybe hinting at some kind of underlying cause? When selecting `Add User` in 2.9.0, the Add New User popup appears with focus over all other UI elements; in 2.10.0, this draws hidden (but with focus) behind the other elements.

Possibly Related 2: The auto-generated parameters for the login form reset every time the Authentication menu is re-opened.

**To Reproduce**
1. Open a Session
2. Get your login form set up by flagging it after a spider run
3. File -> Session Properties
4. Select/Expand Contexts (Use a default one or create/load your own)
5. Select Authentication
 --- > Observe parameter weirdness
6. Set up the form & parameters as needed
7. Select 'Users' in the Contexts sidebar
8. Delete default user
9. Create new user
10. Select this user for all subsequent runs of the scanner and/or spider
11. Start a scan using this user
12. Examine requests being made by scanner
--- > Observe default credentials being fed into parameters (foo-bar@example.com // ZAP )

**Expected behavior**
In 2.9.0: User authenticates successfully; tested pages include both public-facing and those locked behind sign-in.

In 2.10.0: User does NOT authenticate successfully; default credentials for `foo-bar@example.com` are fed into the requests being made by the spider, despite that user being removed from the active context

**Screenshots**
N/A

**Software versions**
 - ZAP: 2.10.0
 - Add-on: N/A
 - OS: Mac OS 10.15.7 -- Have only tested on Mac OS
 - Java: 11.0.1
 - Browser: Chrome Version 88.0.4324.192 (Official Build) (x86_64)

**Errors from the zap.log file**
None - Debug level looks very normal in fact; says it's using my provided credentials, then goes off and does a ton of:

```
2021-03-10 16:48:35,743 [ZAP-SpiderThreadPool-5-thread-1] DEBUG SpiderHtmlFormParser - Existing values: []
2021-03-10 16:48:35,743 [ZAP-SpiderThreadPool-5-thread-1] DEBUG SpiderHtmlFormParser - Generated: foo-bar@example.comFor field user[email]
```

**Additional context**
Reproducible on 2 separate Apple computers; remediation of downgrading to 2.9.0 works on both as well.

**Comment by kingthorin** [comment]
Those log messages aren't authentication they're spider form completion from the formhandler add-on.

**Comment by thc202** [comment]
> Possibly Related 2: The auto-generated parameters for the login form reset every time the Authentication menu is re-opened.

This is most likely the cause since it changes how the login data is formed, it was addressed in #6720.

Closing, it would be great if you could try with a weekly though: https://www.zaproxy.org/download/#weekly
